c2corg_ui/static/js/searchdirective.js

```diff
@@ -0,0 +1,184 @@
+goog.provide('app.SearchController');
+goog.provide('app.searchDirective');
+
+goog.require('app');
+goog.require('app.utils');
+goog.require('ngeo.searchDirective');
+
+
+/**
+ * The directive for the auto-complete search field shown in the header
+ * of every page.
+ *
+ * @return {angular.Directive} Directive Definition Object.
+ */
+app.searchDirective = function() {
+  return {
+    restrict: 'E',
+    controller: 'AppSearchController',
+    bindToController: true,
+    controllerAs: 'ctrl',
+    template:
+        '<input type="text" placeholder="search…" ' +
+        'ngeo-search="ctrl.options" ' +
+        'ngeo-search-datasets="ctrl.datasets" ' +
+        'ngeo-search-listeners="ctrl.listeners">',
+    link:
+        /**
+         * @param {angular.Scope} scope Scope.
+         * @param {angular.JQLite} element Element.
+         * @param {angular.Attributes} attrs Atttributes.
+         */
+        function(scope, element, attrs) {
+          // Empty the search field on focus and blur.
+          element.find('input').on('focus blur', function() {
+            $(this).typeahead('val', '');
+          });
+        }
+  };
+};
+app.module.directive('appSearch', app.searchDirective);
+
+
+
+/**
+ * @constructor
+ * @param {angular.Scope} $rootScope Angular root scope.
+ * @param {angular.$compile} $compile Angular compile service.
+ * @param {string} apiUrl Base URL of the API.
+ * @param {angularGettext.Catalog} gettextCatalog Gettext catalog.
+ * @param {Array.<string>} cultures All cultures.
+ * @ngInject
+ */
+app.SearchController = function(
+    $rootScope, $compile, apiUrl, gettextCatalog, cultures) {
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.apiUrl_ = apiUrl;
+
+  /**
+   * @type {Array.<string>}
+   * @private
+   */
+  this.cultures_ = cultures;
+
+  /**
+   * @type {angularGettext.Catalog}
+   * @private
+   */
+  this.gettextCatalog_ = gettextCatalog;
+
+  /**
+   * @type {TypeaheadOptions}
+   * @export
+   */
+  this.options = /** @type {TypeaheadOptions} */ ({
+    highlight: true,
+    hint: true,
+    minLength: 2
```

Comment:
Do we get useful results with `2`?
Have you considered a bigger number to save resources?